internal/admission/admission.go

```diff
@@ -22,23 +23,21 @@ type OptionsFunc func(*Router) error
 func WithAdmissionHandlers(handlers ...AdmissionHandler) OptionsFunc {
 	return func(r *Router) error {
 		for _, h := range handlers {
-			if _, ok := r.handlers[h.Kind()]; ok {
-				return fmt.Errorf("duplicate handler %s registered", h.Kind())
-			}
-			r.handlers[h.Kind()] = h
+			handlers := r.handlers[h.Kind()]
+			r.handlers[h.Kind()] = append(handlers, h)
 		}
 		return nil
 	}
 }
 
 type Router struct {
-	handlers    map[string]AdmissionHandler
+	handlers    map[string][]AdmissionHandler
```

Comment:
Are we thinking of having an ordered list to look for with a catch-all AllVersionSupported at the end?

This might be an unwarranted optimization but if we use a nested map `map[string]map[string]`, where the inner map stores the supported version string, we wouldn't need to loop over the handlers on L84. But then we'd need to look for a keyword "all" for example to recognize a handler supports all versions. Yeah maybe a bad idea.

Reply:
I don't care about order, we want the first handler that supports the version. If you need to split your handlers for mutation, then it is on the handler author to be explicit about the versions it can support. If there is a reason to split than you clearly cannot support all version and what versions you support is up to your handlers to implement. 

If you had an AllVersionSupported handler and a version specific handler that serve the same Kind, we make no guarantees about which one will handle your request.

Reply:
Gotcha.